projects/plugins/crm/src/automation/class-automation-engine.php

```diff
@@ -0,0 +1,191 @@
+		if ( ! class_exists( $class_name ) ) {
+			throw new Automation_Exception( sprintf( __( 'Step class %s does not exist', 'zero-bs-crm' ), $class_name ), Automation_Exception::STEP_CLASS_NOT_FOUND );
+		}
+		$this->steps_map[ $step_name ] = $class_name;
+	}
+
+	/**
+	 * Get a step class by name
+	 *
+	 * @param string $step_name
+	 * @return string
+	 * @throws Automation_Exception
+	 */
+	public function get_step_class( string $step_name ): string {
+		if ( ! isset( $this->steps_map[ $step_name ] ) ) {
+			throw new Automation_Exception( sprintf( __( 'Step %s does not exist', 'zero-bs-crm' ), $step_name ), Automation_Exception::STEP_CLASS_NOT_FOUND );
+		}
+		return $this->steps_map[ $step_name ];
+	}
+
+	/**
+	 * Add a workflow
+	 *
+	 * @param Automation_Workflow $workflow
+	 * @param bool              $init_workflow
+	 * @return void
+	 * @throws Workflow_Exception
+	 */
+	public function add_workflow( Automation_Workflow $workflow, bool $init_workflow = false ) {
+		$this->workflows[] = $workflow;
+
+		if ( $init_workflow ) {
+			$workflow->init_triggers();
+		}
+	}
+
+	/**
+	 * Build and add a workflow
+	 *
+	 * @param array $workflow_data
+	 * @param bool $init_workflow
+	 * @return Automation_Workflow
+	 * @throws Workflow_Exception
+	 */
+	public function build_add_workflow( array $workflow_data, bool $init_workflow = false ): Automation_Workflow {
+		$workflow = new Automation_Workflow( $workflow_data );
+		$this->add_workflow( $workflow, $init_workflow );
+
+		return $workflow;
+	}
+
+	/**
+	 * Init automation workflows.
+	 *
+	 * @return void
+	 * @throws Workflow_Exception
+	 */
+	public function init_workflows() {
+		
+		/** @var Automation_Workflow $workflow */
+		foreach ( $this->workflows as $workflow ) {
+			$workflow->init_triggers();
+		}
+	}
+
+	/**
+	 * Get step instance
+	 * 
+	 * @param $step_name
+	 * @param array $step_data
+	 * @return Step
+	 * @throws Automation_Exception
+	 */
+	public function get_registered_step( $step_name, array $step_data = array() ): Step {
+		
+		$step_class = $this->get_step_class( $step_name );
+		
+		if ( ! class_exists( $step_class ) ) {
+			throw new Automation_Exception( sprintf( __( 'Step class %s does not exist', 'zero-bs-crm' ), $step_class ), Automation_Exception::STEP_CLASS_NOT_FOUND );
+		}
+		
+		return new $step_class( $step_data );
+	}
+
+	/**
+	 * Get registered steps
+	 * 
+	 * @return array
+	 */
+	public function get_registered_steps():array {
+		return $this->steps_map;
+	}
+
+	/**
+	 * Get trigger instance
+	 *
+	 * @param string $trigger_name
+	 * @return string
+	 * @throws Automation_Exception
+	 */
+	public function get_trigger_class( string $trigger_name ): string {
+		
+		if ( ! isset( $this->triggers_map[ $trigger_name ] ) ) {
+			throw new Automation_Exception( sprintf( __( 'Trigger %s does not exist', 'zero-bs-crm' ), $trigger_name ), Automation_Exception::TRIGGER_CLASS_NOT_FOUND );
+		}
+		
+		return $this->triggers_map[ $trigger_name ];
+	}
```

Comment:
This method doesn't seem to actually do what the DocBlock says but IMO it would make sense to refactor it to do so since it would simplify how it's being used in the code:

```suggestion
	public function get_trigger( string $trigger_name ): Trigger {

		if ( ! isset( $this->triggers_map[ $trigger_name ] ) ) {
			throw new Automation_Exception( sprintf( __( 'Trigger %s does not exist', 'zero-bs-crm' ), $trigger_name ), Automation_Exception::TRIGGER_CLASS_NOT_FOUND );
		}

		return new $this->triggers_map[ $trigger_name ]();
	}
```

That way this implementation:

```php
$class_name = $this->automation_engine->get_trigger_class( $trigger_name );

/** @var Trigger $trigger */
$trigger = new $class_name();
$trigger->init( $this );
```

would look like this instead:

```php
$trigger = $this->automation_engine->get_trigger( $trigger_name );
$trigger->init( $this );
```